### DISCUSSION TITLE
Confirmation on production readiness of diesel-async currently?

### DISCUSSION THREAD

**Discussion by AbdulWMughal** [weiznich/diesel_async#166] [Q&A]
Dear Weiznich & Co,

1) On the initial release of this project you stated it wasn’t production ready. Can you confirm whether as of today the diesel-async crate is production ready?

2) Do you know if diesel performs better/quicker than executing raw SQL directly through a tokio_postgres client connection without using an ORM or query builder?

well done on your work so far i wish to see rust reach and surpass the level of production ready libraries and tools that other older more established languages (python, etc) have. Especially a pytorch equivalent.

lastly, could you link any documentation that lists all the new async methods from their old sync methods. Like execute.

Kind Regards

**Comment by weiznich** [comment]
> On the initial release of this project you stated it wasn’t production ready. Can you confirm whether as of today the diesel-async crate is production ready?

I cannot answer your question about whether this library is "production ready" or not. That depends on your requirements alone. You need to take a look at the code and it's limitations and see if it fits your requirements or not. That's nothing someone else can do for you.

> Do you know if diesel performs better/quicker than executing raw SQL directly through a tokio_postgres client connection without using an ORM or query builder?

Performance is another matter that depends on your requirements. You shouldn't trust any external provided benchmarks if you really care about performance. You need to measure your own workloads there instead of relaying on something generic. That written, there are some benchmark results here: https://github.com/diesel-rs/metrics/

> lastly, could you link any documentation that lists all the new async methods from their old sync methods. Like execute.

There exists no such document yet, but generally speaking all the methods on `diesel::RunQueryDsl` exists 1:1 in `diesel_async::RunQueryDsl`, so you would just change that one import and add `.await` to the relevant method calls.

Finally: Our issue tracker is no support forum. Please do not use it to ask support questions. I've converted your issue to a discussion this time. For the next questions please use directly the support forum, or your issues will be closed as invalid.

**Reply by AbdulWMughal** [reply]
Looking at the benchmark metrics i was shocked at how poor tokio_Postgres performed. Any ideas as to why? For medium complex queries, inserts, associations it seemed to do poorly with large load of queries. This has me baffled. Perhaps manually creating a transaction and using raw versions (query_raw) may be the solution. What do you think? 

https://github.com/diesel-rs/metrics/
